### PR TITLE
Support imperative learning rate scheduler

### PR DESCRIPTION
**Pull request by velconia**
1. move imperative mnist ut to test_imperative_mnist
2. add optimizer lr scheduler ut into test_imperative_optimizer.py
3. implement lr scheduler in imperative mode

### CODE REVIEW

File: python/paddle/fluid/imperative/learning_rate_scheduler.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# limitations under the License.
+
+from __future__ import print_function
+
+from .. import unique_name
+
+__all__ = ['PiecewiseDecay']
+
+
+class LearningRateDecay(object):
```
Comment (panyx0718) [inline python/paddle/fluid/imperative/learning_rate_scheduler.py]: does this support static graph? why not use existing lr_scheduler?

Comment (velconia) [reply python/paddle/fluid/imperative/learning_rate_scheduler.py]: static lr and dynamic lr is used via interface in ```learning_rate_scheduler.py``` under the fluid dir

File: python/paddle/fluid/optimizer.py [hunk 2]
```diff
@@ -76,24 +87,43 @@ def get_opti_var_name_list(self):
...
     def _create_global_learning_rate(self):
-        lr = self._global_learning_rate()
-
-        if isinstance(lr, framework.Variable):
-            return
+        if imperative_base.enabled():
+            # create learning rate Variable
+            if isinstance(self._learning_rate, float):
+                self._learning_rate_map[framework.default_main_program(
+                )] = layers.create_global_var(
```
Comment (panyx0718) [inline python/paddle/fluid/optimizer.py]: can this be created many times?

File: python/paddle/fluid/optimizer.py [hunk 3]
```diff
@@ -76,24 +87,43 @@ def get_opti_var_name_list(self):
...
+                    name=unique_name.generate("learning_rate"),
+                    shape=[1],
+                    value=float(self._learning_rate),
+                    dtype='float32' if self._dtype is None else self._dtype,
+                    persistable=True)
+            # get learning rate Variable from LearningRateDecay
+            elif isinstance(self._learning_rate, LearningRateDecay):
+                self._learning_rate_map[framework.default_main_program(
```
Comment (panyx0718) [inline python/paddle/fluid/optimizer.py:111]: why use main program?

File: python/paddle/fluid/optimizer.py (hunk 2 above)
Comment (velconia) [reply python/paddle/fluid/optimizer.py]: no, this should only be called once Optimizer.__init__

File: python/paddle/fluid/optimizer.py (hunk 3 above)
Comment (velconia) [reply python/paddle/fluid/optimizer.py:111]: to keep the same with static mode code

File: python/paddle/fluid/optimizer.py (hunk 2 above)
Comment (velconia) [reply python/paddle/fluid/optimizer.py]: Done

**Review by panyx0718 — APPROVED** [review]
